File: go/lc/lc101.go
package main

func isSymmetric(root *TreeNode) bool {
	if root == nil {
		return true
	}
	var cmp func(l, r *TreeNode) bool
	cmp = func(l, r *TreeNode) bool {
		if l == nil && r == nil {
			return true
		} else if l != nil && r != nil {
			if l.Val != r.Val {
				return false
			}
			return cmp(l.Left, r.Right) && cmp(l.Right, r.Left)
		} else {
			return false
		}
	}
	return cmp(root.Left, root.Right)
}

File: java/com/lc/lc101.java
package lc;

import java.util.*;

public class lc101 {
    public boolean isSymmetric(TreeNode root) {
        return check(root, root);
    }

    public boolean check(TreeNode l, TreeNode r) {
        if (l == null && r == null) {
            return true;
        } else if (l == null || r == null) {
            return false;
        }

        return l.val == r.val && check(l.left, r.right) && check(l.right, r.left);
    }
}
